pkg/log/log.go

```diff
@@ -108,3 +112,42 @@ func parseLogLevel(logLevel string) (level.Option, error) {
 		return nil, fmt.Errorf("unrecognized log level %q", logLevel)
 	}
 }
+
+const logOnceTimedDuration = time.Minute
+
+// timedLogger logs from logStore every logOnceTimedDuration. It deletes the log entry from the store
+// after it has logged once.
+func timedLogger() {
+	applyKind := func(logMsg []interface{}) (newLogMsg []interface{}) {
+		newLogMsg = append(newLogMsg, "kind", "rate-limited", "duration", logOnceTimedDuration.String())
+		newLogMsg = append(newLogMsg, logMsg...)
+		return
+	}
+	time.Sleep(logOnceTimedDuration)
+	logStoreMux.Lock()
+	defer logStoreMux.Unlock()
+	for k, logMsg := range logStore {
+		Debug(applyKind(logMsg)...)
+		delete(logStore, k)
+	}
+	if len(logStore) != 0 {
+		panic(fmt.Sprintf("logStore not empty after flushing rate-limited logs: expected size to be 0 but received %d", len(logStore)))
+	}
+	// delete() does not remove the value from the map. It just sets the flag, meaning memory will increase even though less (at first).
+	// Hence, better to re-initialize it within the lock.
+	logStore = make(map[string][]interface{})
```

Comment:
if we do this then we don't need the delete above (delete can get expensive), Or the panic. I like that more.

Reply:
>  I like that more

If we just do the delete, then we aren't clearing the memory used by the map for storing log. That's the behaviour of delete() in map. Hence, I reinitialized it.

Reply:
I think, going with just initialization should be better way here in terms of efficiency and memory. Do you agree?

Reply:
sorry I meant that I like the initialization more - we should remove the delete and the panic

Reply:
oh, got it. The current was in line with that method. No prob.